Fix for torch.save not saving source files

chauhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28965 Fix for torch.save not saving source files**

Summary: Fixed the reference to correct object

Test Plan:
    Added new unit test test_serialization_save_warnings in test_torch
    Verified by running the test_torch tests
    
Reviewers: vincentqb zou3519

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

Differential Revision: [D18306797](https://our.internmc.facebook.com/intern/diff/D18306797)

Commit:
Fix for torch.save not saving source files

Summary: Fixed the reference to correct object

Test Plan:
Verified by running the test_torch tests and a sample test case to see file getting saved with no errors or warnings
import torch
m = torch.nn.Linear(500, 10)
torch.save(m, './test.pth')

Reviewers: vincentqb

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]

chauhang commented:
@pytorchbot retest this please

Commit:
Update on "Fix for torch.save not saving source files"


Summary: Fixed the reference to correct object

Test Plan:
Verified by running the test_torch tests and a sample test case to see file getting saved with no errors or warnings
import torch
m = torch.nn.Linear(500, 10)
torch.save(m, './test.pth')

Reviewers: vincentqb

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]

Commit:
Update on "Fix for torch.save not saving source files"


Summary: Fixed the reference to correct object

Test Plan:
    Added new unit test test_serialization_save_warnings in test_torch
    Verified by running the test_torch tests
    
Reviewers: vincentqb zou3519

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]

chauhang commented:
@pytorchbot retest this please

chauhang commented:
@pytorchbot retest this please

Commit:
Update on "Fix for torch.save not saving source files"


Summary: Fixed the reference to correct object

Test Plan:
    Added new unit test test_serialization_save_warnings in test_torch
    Verified by running the test_torch tests
    
Reviewers: vincentqb zou3519

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]

Commit:
Update on "Fix for torch.save not saving source files"


Summary: Fixed the reference to correct object

Test Plan:
    Added new unit test test_serialization_save_warnings in test_torch
    Verified by running the test_torch tests
    
Reviewers: vincentqb zou3519

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]

zou3519 commented on test/test_torch.py:
```diff
@@ -4117,9 +4117,18 @@ def __reduce__(self):
+    def test_serialization_save_warnings(self):
+        with warnings.catch_warnings(record=True) as warns:
+            with tempfile.NamedTemporaryFile() as checkpoint:
+                x = torch.save(torch.nn.Module(), checkpoint)
```
I'm not sure that one is supposed to instantiate a nn.Module() directly like this. Could we change this to `nn.Linear(2, 3)` to better reflect how a user would use the API?

zou3519 commented on test/test_torch.py:
```diff
@@ -4117,9 +4117,18 @@ def __reduce__(self):
+        with warnings.catch_warnings(record=True) as warns:
+            with tempfile.NamedTemporaryFile() as checkpoint:
+                x = torch.save(torch.nn.Module(), checkpoint)
+                self.assertEquals(len(warns), 0)
```
Does this actually work? I thought that you have to check `len(warns)` outside the context manager scope.

Does the test fail without your changes? If not, then the test won't help catch a future regression

zou3519 replied:
Okay, I read up on the docs and this should work. Please ignore my comment

zou3519 reviewed (commented):
This almost looks good, I had some questions

Commit:
Update on "Fix for torch.save not saving source files"


Summary: Fixed the reference to correct object

Test Plan:
    Added new unit test test_serialization_save_warnings in test_torch
    Verified by running the test_torch tests
    
Reviewers: vincentqb zou3519

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]

Commit:
Update on "Fix for torch.save not saving source files"


Summary: Fixed the reference to correct object

Test Plan:
    Added new unit test test_serialization_save_warnings in test_torch
    Verified by running the test_torch tests
    
Reviewers: vincentqb zou3519

Subscribers: gchauhan

Tasks:

Tags:

Fix for torch.save not saving file

Fix for torch.save not working

[ghstack-poisoned]
